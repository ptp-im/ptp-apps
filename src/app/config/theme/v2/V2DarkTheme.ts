import color from 'color';

import type { MD2Theme } from 'react-native-paper';
import { black, red800, white } from './colors';
import { V2LightTheme } from './V2LightTheme';

export const V2DarkTheme: MD2Theme = {
  ...V2LightTheme,
  dark: true,
  mode: 'adaptive',
  version: 2,
  isV3: false,
  colors: {
    ...V2LightTheme.colors,
    // primary: '#BB86FC',
    accent: '#03dac6',
    // background: '#121212',
    surface: '#121212',
    error: '#CF6679',
    onSurface: '#FFFFFF',
    // text: white,
    disabled: color(white).alpha(0.38).rgb().string(),
    placeholder: color(white).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA100,

    msgUnreadBadge: red800,
    primary: 'rgb(10, 132, 255)',
    background: 'rgb(1, 1, 1)',
    card: 'rgb(18, 18, 18)',
    text: 'rgb(229, 229, 231)',
    textPrimary: 'rgb(10, 132, 255)',
    border: 'rgb(39, 39, 41)',
    notification: 'rgb(255, 69, 58)',
  },
};

export const V2NavDarkColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
