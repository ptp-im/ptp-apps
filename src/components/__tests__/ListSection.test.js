import * as React from 'react';
import { StyleSheet } from 'react-native';

import renderer from 'react-test-renderer';

import { red500 } from '../../styles/themes/v2/colors';
import ListIcon from '../List/ListIcon.tsx';
import ListItem from '../List/ListItem.tsx';
import ListSection from '../List/ListSection.tsx';
import ListSubheader from '../List/ListSubheader.tsx';

const styles = StyleSheet.create({
  itemColor: {
    color: red500,
  },
});

it('renders list section without subheader', () => {
  const tree = renderer
    .create(
      <ListSection>
        <ListItem
          title='First Item'
          left={(props) => <ListIcon {...props} icon='folder' />}
        />
        <ListItem
          title='Second Item'
          left={(props) => <ListIcon {...props} icon='folder' />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list section with subheader', () => {
  const tree = renderer
    .create(
      <ListSection>
        <ListSubheader>Some title</ListSubheader>
        <ListItem
          title='First Item'
          left={(props) => <ListIcon {...props} icon='folder' />}
        />
        <ListItem
          title='Second Item'
          left={(props) => <ListIcon {...props} icon='folder' />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list section with custom title style', () => {
  const tree = renderer
    .create(
      <ListSection title='Some title' titleStyle={styles.itemColor}>
        <ListItem
          title='First Item'
          left={(props) => <ListIcon {...props} icon='folder' />}
        />
        <ListItem
          title='Second Item'
          left={(props) => <ListIcon {...props} icon='folder' />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
