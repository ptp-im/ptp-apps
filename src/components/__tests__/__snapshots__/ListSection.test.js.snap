// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list section with custom title style 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "animation": Object {
        "scale": 1,
      },
      "colors": Object {
        "backdrop": "rgba(50, 47, 55, 0.4)",
        "background": "rgba(255, 251, 254, 1)",
        "elevation": Object {
          "level0": "transparent",
          "level1": "rgb(247, 243, 249)",
          "level2": "rgb(243, 237, 246)",
          "level3": "rgb(238, 232, 244)",
          "level4": "rgb(236, 230, 243)",
          "level5": "rgb(233, 227, 241)",
        },
        "error": "rgba(179, 38, 30, 1)",
        "errorContainer": "rgba(249, 222, 220, 1)",
        "inverseOnSurface": "rgba(244, 239, 244, 1)",
        "inversePrimary": "rgba(208, 188, 255, 1)",
        "inverseSurface": "rgba(49, 48, 51, 1)",
        "onBackground": "rgba(28, 27, 31, 1)",
        "onError": "rgba(255, 255, 255, 1)",
        "onErrorContainer": "rgba(65, 14, 11, 1)",
        "onPrimary": "rgba(255, 255, 255, 1)",
        "onPrimaryContainer": "rgba(33, 0, 93, 1)",
        "onSecondary": "rgba(255, 255, 255, 1)",
        "onSecondaryContainer": "rgba(29, 25, 43, 1)",
        "onSurface": "rgba(28, 27, 31, 1)",
        "onSurfaceDisabled": "rgba(28, 27, 31, 0.38)",
        "onSurfaceVariant": "rgba(73, 69, 79, 1)",
        "onTertiary": "rgba(255, 255, 255, 1)",
        "onTertiaryContainer": "rgba(49, 17, 29, 1)",
        "outline": "rgba(121, 116, 126, 1)",
        "outlineVariant": "rgba(202, 196, 208, 1)",
        "primary": "rgba(103, 80, 164, 1)",
        "primaryContainer": "rgba(234, 221, 255, 1)",
        "scrim": "rgba(0, 0, 0, 1)",
        "secondary": "rgba(98, 91, 113, 1)",
        "secondaryContainer": "rgba(232, 222, 248, 1)",
        "shadow": "rgba(0, 0, 0, 1)",
        "surface": "rgba(255, 251, 254, 1)",
        "surfaceDisabled": "rgba(28, 27, 31, 0.12)",
        "surfaceVariant": "rgba(231, 224, 236, 1)",
        "tertiary": "rgba(125, 82, 96, 1)",
        "tertiaryContainer": "rgba(255, 216, 228, 1)",
      },
      "dark": false,
      "fonts": Object {
        "bodyLarge": Object {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "bodyMedium": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        "bodySmall": Object {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "lineHeight": 16,
        },
        "displayLarge": Object {
          "fontFamily": "System",
          "fontSize": 57,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 64,
        },
        "displayMedium": Object {
          "fontFamily": "System",
          "fontSize": 45,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 52,
        },
        "displaySmall": Object {
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 44,
        },
        "headlineLarge": Object {
          "fontFamily": "System",
          "fontSize": 32,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 40,
        },
        "headlineMedium": Object {
          "fontFamily": "System",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
        },
        "headlineSmall": Object {
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        "labelLarge": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        "labelMedium": Object {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "labelSmall": Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "titleLarge": Object {
          "fontFamily": "System",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        "titleMedium": Object {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "titleSmall": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
      },
      "isV3": true,
      "roundness": 4,
      "version": 3,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 13,
          },
          Object {
            "color": "rgba(73, 69, 79, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          Object {
            "color": "#f44336",
          },
        ],
      ]
    }
  >
    Some title
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
              "marginVertical": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          â–¡
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                undefined,
              ],
            ]
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
              "marginVertical": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          â–¡
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                undefined,
              ],
            ]
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list section with subheader 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "animation": Object {
        "scale": 1,
      },
      "colors": Object {
        "backdrop": "rgba(50, 47, 55, 0.4)",
        "background": "rgba(255, 251, 254, 1)",
        "elevation": Object {
          "level0": "transparent",
          "level1": "rgb(247, 243, 249)",
          "level2": "rgb(243, 237, 246)",
          "level3": "rgb(238, 232, 244)",
          "level4": "rgb(236, 230, 243)",
          "level5": "rgb(233, 227, 241)",
        },
        "error": "rgba(179, 38, 30, 1)",
        "errorContainer": "rgba(249, 222, 220, 1)",
        "inverseOnSurface": "rgba(244, 239, 244, 1)",
        "inversePrimary": "rgba(208, 188, 255, 1)",
        "inverseSurface": "rgba(49, 48, 51, 1)",
        "onBackground": "rgba(28, 27, 31, 1)",
        "onError": "rgba(255, 255, 255, 1)",
        "onErrorContainer": "rgba(65, 14, 11, 1)",
        "onPrimary": "rgba(255, 255, 255, 1)",
        "onPrimaryContainer": "rgba(33, 0, 93, 1)",
        "onSecondary": "rgba(255, 255, 255, 1)",
        "onSecondaryContainer": "rgba(29, 25, 43, 1)",
        "onSurface": "rgba(28, 27, 31, 1)",
        "onSurfaceDisabled": "rgba(28, 27, 31, 0.38)",
        "onSurfaceVariant": "rgba(73, 69, 79, 1)",
        "onTertiary": "rgba(255, 255, 255, 1)",
        "onTertiaryContainer": "rgba(49, 17, 29, 1)",
        "outline": "rgba(121, 116, 126, 1)",
        "outlineVariant": "rgba(202, 196, 208, 1)",
        "primary": "rgba(103, 80, 164, 1)",
        "primaryContainer": "rgba(234, 221, 255, 1)",
        "scrim": "rgba(0, 0, 0, 1)",
        "secondary": "rgba(98, 91, 113, 1)",
        "secondaryContainer": "rgba(232, 222, 248, 1)",
        "shadow": "rgba(0, 0, 0, 1)",
        "surface": "rgba(255, 251, 254, 1)",
        "surfaceDisabled": "rgba(28, 27, 31, 0.12)",
        "surfaceVariant": "rgba(231, 224, 236, 1)",
        "tertiary": "rgba(125, 82, 96, 1)",
        "tertiaryContainer": "rgba(255, 216, 228, 1)",
      },
      "dark": false,
      "fonts": Object {
        "bodyLarge": Object {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "bodyMedium": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        "bodySmall": Object {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "lineHeight": 16,
        },
        "displayLarge": Object {
          "fontFamily": "System",
          "fontSize": 57,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 64,
        },
        "displayMedium": Object {
          "fontFamily": "System",
          "fontSize": 45,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 52,
        },
        "displaySmall": Object {
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 44,
        },
        "headlineLarge": Object {
          "fontFamily": "System",
          "fontSize": 32,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 40,
        },
        "headlineMedium": Object {
          "fontFamily": "System",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
        },
        "headlineSmall": Object {
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        "labelLarge": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        "labelMedium": Object {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "labelSmall": Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "titleLarge": Object {
          "fontFamily": "System",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        "titleMedium": Object {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "titleSmall": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
      },
      "isV3": true,
      "roundness": 4,
      "version": 3,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 13,
          },
          Object {
            "color": "rgba(73, 69, 79, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          undefined,
        ],
      ]
    }
  >
    Some title
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
              "marginVertical": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          â–¡
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                undefined,
              ],
            ]
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
              "marginVertical": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          â–¡
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                undefined,
              ],
            ]
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list section without subheader 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "animation": Object {
        "scale": 1,
      },
      "colors": Object {
        "backdrop": "rgba(50, 47, 55, 0.4)",
        "background": "rgba(255, 251, 254, 1)",
        "elevation": Object {
          "level0": "transparent",
          "level1": "rgb(247, 243, 249)",
          "level2": "rgb(243, 237, 246)",
          "level3": "rgb(238, 232, 244)",
          "level4": "rgb(236, 230, 243)",
          "level5": "rgb(233, 227, 241)",
        },
        "error": "rgba(179, 38, 30, 1)",
        "errorContainer": "rgba(249, 222, 220, 1)",
        "inverseOnSurface": "rgba(244, 239, 244, 1)",
        "inversePrimary": "rgba(208, 188, 255, 1)",
        "inverseSurface": "rgba(49, 48, 51, 1)",
        "onBackground": "rgba(28, 27, 31, 1)",
        "onError": "rgba(255, 255, 255, 1)",
        "onErrorContainer": "rgba(65, 14, 11, 1)",
        "onPrimary": "rgba(255, 255, 255, 1)",
        "onPrimaryContainer": "rgba(33, 0, 93, 1)",
        "onSecondary": "rgba(255, 255, 255, 1)",
        "onSecondaryContainer": "rgba(29, 25, 43, 1)",
        "onSurface": "rgba(28, 27, 31, 1)",
        "onSurfaceDisabled": "rgba(28, 27, 31, 0.38)",
        "onSurfaceVariant": "rgba(73, 69, 79, 1)",
        "onTertiary": "rgba(255, 255, 255, 1)",
        "onTertiaryContainer": "rgba(49, 17, 29, 1)",
        "outline": "rgba(121, 116, 126, 1)",
        "outlineVariant": "rgba(202, 196, 208, 1)",
        "primary": "rgba(103, 80, 164, 1)",
        "primaryContainer": "rgba(234, 221, 255, 1)",
        "scrim": "rgba(0, 0, 0, 1)",
        "secondary": "rgba(98, 91, 113, 1)",
        "secondaryContainer": "rgba(232, 222, 248, 1)",
        "shadow": "rgba(0, 0, 0, 1)",
        "surface": "rgba(255, 251, 254, 1)",
        "surfaceDisabled": "rgba(28, 27, 31, 0.12)",
        "surfaceVariant": "rgba(231, 224, 236, 1)",
        "tertiary": "rgba(125, 82, 96, 1)",
        "tertiaryContainer": "rgba(255, 216, 228, 1)",
      },
      "dark": false,
      "fonts": Object {
        "bodyLarge": Object {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "bodyMedium": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        "bodySmall": Object {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "lineHeight": 16,
        },
        "displayLarge": Object {
          "fontFamily": "System",
          "fontSize": 57,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 64,
        },
        "displayMedium": Object {
          "fontFamily": "System",
          "fontSize": 45,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 52,
        },
        "displaySmall": Object {
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 44,
        },
        "headlineLarge": Object {
          "fontFamily": "System",
          "fontSize": 32,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 40,
        },
        "headlineMedium": Object {
          "fontFamily": "System",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
        },
        "headlineSmall": Object {
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        "labelLarge": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        "labelMedium": Object {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "labelSmall": Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "titleLarge": Object {
          "fontFamily": "System",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        "titleMedium": Object {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        "titleSmall": Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
      },
      "isV3": true,
      "roundness": 4,
      "version": 3,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
              "marginVertical": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          â–¡
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                undefined,
              ],
            ]
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
              "marginVertical": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          â–¡
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                undefined,
              ],
            ]
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;
